app/javascript/components/Survey.jsx

```diff
@@ -63,6 +63,23 @@ class Survey extends Component {
       formIcon = (<FontIcon>file_upload</FontIcon>)
     }
 
+    const addTrainingSentence = () => {
```

Comment:
Move this three definitions to the KeywordInput component. Make it so it receives the `updateConfig('training_sentences')` function and uses it to trigger the updates